crates/rover-client/src/operations/persisted_queries/publish/types.rs

```diff
@@ -107,6 +110,87 @@ impl From<PersistedQueriesPublishInput> for QueryVariables {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RelayPersistedQueryManifest {
+    #[serde(flatten)]
+    operations: HashMap<String, String>,
+}
+
+impl TryFrom<RelayPersistedQueryManifest> for ApolloPersistedQueryManifest {
+    type Error = RoverClientError;
+
+    fn try_from(relay_manifest: RelayPersistedQueryManifest) -> Result<Self, Self::Error> {
+        let mut operations_with_invalid_type = Vec::new();
+        let mut anonymous_operations = Vec::new();
+        let mut operations = Vec::new();
+        for (id, body) in relay_manifest.operations {
+            let ast = Parser::new(&body).parse();
```

Comment:
While I don't think we should run full-on validations, we probably should at least check that the parsing that we're doing didn't fail. In Router it looks like we usually check `ast.errors().peekable().peek().is_some()`. That way if the string doesn't parse, we can get a syntax error instead of an "it's anonymous" error which would be misleading.

Reply:
Would we want to report those syntax errors? I purposely did not include this because if we want to report syntax errors we should just be using the compiler like my previous commit.

Reply:
(Oops, I responded to your similar top-level comment at the top level instead.)

Reply:
OK, I guess apollo-parser is a lot more capable of dealing with syntax errors than I thought.  For example, if we take the NewsfeedQuery body in `it_can_convert_relay_manifest_to_apollo_manifest` and add an unpaired `}` to either the beginning or the end of the body, the test still passes: while that `}` causes a syntax error (which you don't check for), it still manages to parse the operation and get it into the stuff that your code looks for.

On the other hand, adding an unpaired `{` to either the beginning or end of the body leads to a `The following operation IDs do not have a name` error, since I guess that's enough to make the parser not find the operation. (A bit surprising that unpaired `}` at the end is "ok" but unpaired `{` at the end is not.

I don't know how I feel about this. Personally it feels a little weird that some syntax errors will result in successful execution and some syntax errors will result in `The following operation IDs do not have a name`, and no syntax errors will result in "hey there's a syntax error". If we're relying on the outcome of parsing, then it seems like we should provide the most precise reason that parsing didn't succeed?

I see that the `apollo-parser` docs say that its parsing is error-resilient, but is there a *detailed* description of what that means that would be sufficient to give a mental model that explains the observations above about the curly braces?  Dealing with the data structures created by parsing code that doesn't parse properly just feels very complex to me and hard to wrap my head around. If we really want to be able to process syntactically invalid GraphQL then we might as well just switch to using parser::Lexer, reading the first two tokens, asserting they are both names and that the first one is query/mutation/subscription, and moving on...

In practice hopefully none of this error handling talk matters much anyway because in practice Relay Compiler is not actually spitting out broken code :)